modules/pam_unix/passverify.c

```diff
@@ -266,8 +266,10 @@ PAMH_ARG_DECL(int check_shadow_expiry,
 	    && (curdays - spent->sp_lstchg > spent->sp_inact)
 	    && (curdays - spent->sp_lstchg > spent->sp_max + spent->sp_inact)
 	    && (spent->sp_max != -1) && (spent->sp_inact != -1))
-	    || (crypt_checksalt(spent->sp_pwdp) == CRYPT_SALT_METHOD_DISABLED)
-	    || (crypt_checksalt(spent->sp_pwdp) == CRYPT_SALT_INVALID)) {
+	    || (((crypt_checksalt(spent->sp_pwdp) == CRYPT_SALT_METHOD_DISABLED)
+		 || (crypt_checksalt(spent->sp_pwdp) == CRYPT_SALT_INVALID))
+	        && ((spent->sp_pwdp != NULL)
```

Comment:
Can you please factor this check for locked out passwords to a static function? And should there be also check for empty password hash? spent->sp_pwdp[0] == '\0' Another question is whether crypt_checksalt(spent->sp_pwdp) == CRYPT_SALT_INVALID should actually return the error. Most probably it should not.

Reply:
The more I think about it, the more I am inclined to drop this part of the merged commit completely. It does not belong here.